src/Server_TEST.cc

```diff
@@ -475,6 +475,100 @@ TEST_P(ServerFixture, RunNonBlocking)
   EXPECT_FALSE(*server.Running(0));
 }
 
+/////////////////////////////////////////////////
+TEST_P(ServerFixture, RunOnceUnpaused)
+{
+  gazebo::Server server;
+  EXPECT_FALSE(server.Running());
+  EXPECT_FALSE(*server.Running(0));
+  EXPECT_EQ(0u, *server.IterationCount());
+
+  // Load a system
+  gazebo::SystemLoader systemLoader;
+  auto mockSystemPlugin = systemLoader.LoadPlugin(
+      "libMockSystem.so", "ignition::gazebo::MockSystem", nullptr);
+  ASSERT_TRUE(mockSystemPlugin.has_value());
+
+  // Check that it was loaded
+  const size_t systemCount = *server.SystemCount();
+  EXPECT_TRUE(*server.AddSystem(mockSystemPlugin.value()));
+  EXPECT_EQ(systemCount + 1, *server.SystemCount());
+
+  // Query the interface from the plugin
+  auto system = mockSystemPlugin.value()->QueryInterface<gazebo::System>();
+  EXPECT_NE(system, nullptr);
+  auto mockSystem = dynamic_cast<gazebo::MockSystem*>(system);
+  EXPECT_NE(mockSystem, nullptr);
+
+  // No steps should have been executed
+  EXPECT_EQ(0u, mockSystem->preUpdateCallCount);
+  EXPECT_EQ(0u, mockSystem->updateCallCount);
+  EXPECT_EQ(0u, mockSystem->postUpdateCallCount);
+
+  // Make the server run fast
+  server.SetUpdatePeriod(1ns);
+
+  while (*server.IterationCount() < 100)
+    server.RunOnce(false);
+
+  // Check that the server provides the correct information
+  EXPECT_EQ(*server.IterationCount(), 100);
```

Comment:
Compiler warning:

```
/var/lib/jenkins/workspace/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/ign-gazebo/src/Server_TEST.cc:515:3:   required from here
/var/lib/jenkins/workspace/ignition_gazebo-ci-pr_any-ubuntu_auto-amd64/ign-gazebo/test/gtest/include/gtest/gtest.h:1448:16: warning: comparison of integer expressions of different signedness: ‘const long unsigned int’ and ‘const int’ [-Wsign-compare]
```

Reply:
Fixed in https://github.com/ignitionrobotics/ign-gazebo/compare/db2989e16fb84ac77780d252a7d75501013bf9be..e991e58e4be16b667f551c2381f0b7ed37e9446c